students/jesse_miller/session02/print_grid.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/local/bin/python3
+
+def do_two(f):
+    """This will execute two of the give function"""
+    f()
+    f()
+
+def do_four(f):
+    """This in turn executes four"""
+    do_two(f)
+    do_two(f)
+
+def print_line():
+    """Definition of the lines"""
+    print('+ - - - - + - - - - +'),
+
+def print_post():
+    """Definition of the columns"""
+    print('|         |         |'),
+
+def print_row():
+    """This is where the magic starts.  It prints one line and four columns"""
+    print_line()
+    do_four(print_post)
+
+def print_grid():
+    """The remainder of the magic.  It doubles the above function and prints the last row"""
+    do_two(print_row)
+    print_line()
```

Comment:
This is a bit to complex to follow and difficult to read and understand what is going on.  Take a look at `for` loops which I think would help you achieve what you are trying to do here.
```
print_line()
for line in range(3):
    for post in range(3):
        print_post()
    print_line()
```
see if you can update this file to reflect using for loops

Reply:
I will try, I'm not certain I can get it done by tonight but I will do what I can.